src/proxy/ClientTable.php

```diff
@@ -40,14 +47,36 @@ public function __construct(ClientBuild $build, array $data, array $config = [])
         $this->build = $build;
         $this->_data = $data;
         parent::__construct($config);
-    }
 
+        if (!$this->db) {
+            $this->setDb(Yii::$app->db);
+        }
+    }
+    
+    /**
+     * @return Connection|null
+     * @since 2.0.0
+     */
+    public function getDb()
+    {
+        return $this->db;
```

Comment:
Instead of constructor we should do this check here, so $db (or getDb()) will always return a valid result:

```php
public function getDb()
{
    return $this->_db ? $this->_db : Yii::$app->db;
}
```

Reply:
Usually we also write directly the private variable above the getter/setter methods. 

As we implement BaseObject we need a @property Connection $db inside the class. Thanks